Content/preparing for the merge hear from our truffle engineers (#1278)
trufflesuite/trufflesuite.com: src/blog/preparing-for-the-merge-hear-from-our-truffle-engineers/index.md

---
title: Preparing for the Merge. Hear from our Truffle engineers 
hide:
  - navigation
---

**By [Kingsley Arinze](https://twitter.com/heydamali)**

In a previous [blog post](https://trufflesuite.com/blog/the-merge-and-what-it-means-for-truffle), we discussed the Merge in detail, including how it upgrades Ethereum as we know it today. Check out the post to better understand and prepare for the Merge.

> Here, we expand on what the Merge means for us at Truffle and what we're doing to prepare for it. We spoke with our resident L2 experts, [Kevin Weaver](https://twitter.com/kevin_weaver), Senior Blockchain Engineer, and [Faina Shalts](https://twitter.com/bitsplaining), Lead Blockchain Engineer, and Lead Blockchain Engineer [David Murdoch](https://twitter.com/atdavidmurdoch), who leads the Ganache team at Truffle.

Reviewer: Just adding some parallel structure
```suggestion
Here, we expand on what the Merge means for us at Truffle and what we're doing to prepare for it. We spoke with our resident L2 experts, [Kevin Weaver](https://twitter.com/kevin_weaver), Senior Blockchain Engineer, and [Faina Shalts](https://twitter.com/bitsplaining), Lead Blockchain Engineer, and our Lead Blockchain Engineer [David Murdoch](https://twitter.com/atdavidmurdoch), who leads the Ganache team at Truffle.
```

### Is there a set date for the Merge to happen?

**Kevin:**

*“The tentative date for the Merge is sometime during the week of Sept. 19, 2022. However, this date is subject to change based on the outcome of the Goerli testnet merge.”*

### What do Truffle users need to know/do ahead of the Merge?

**David Murdoch, Lead Blockchain Engineer, Truffle:**

*“They probably don't need to do anything unless they run a node and/or validator. Everything will continue to work at the dApp layer as it does now. There are some changes, but 99.99% of use cases won't care or notice.”* 

**Kevin:**

*“Trufflers do not need to take any actions to prepare for the Merge. The switch to PoS will involve changes at the level of the consensus mechanism, which underpins the application layer, but not on the application layer itself.”*

### What changes should users expect from Ganache?

David on possible upgrades to the Ganache local Ethereum simulator:

> *“We will need to add a few new block tags. We currently use ‘latest’, ‘earliest’ and ‘pending.’, Soon, we may add ‘safe’ ‘finalized’, and maybe ‘unsafe.’ What those mean exactly is TBD). Maybe a new block header field and the difficulty field's behavior will change. Almost everything else will be internal changes mostly in the EVM layer. Some advanced users may care about some subtle changes to OPCODE behavior, but those users will be few and far between.”*

Reviewer 2: ```suggestion
*“We will need to add a few new block tags. We currently use `"latest"`, `"earliest"`, and `"pending"`. We may add `"safe"`, `"finalized"`, and probably `"unsafe"` to this list, though we aren't yet sure the what significance it will have outside of forking. The "difficulty" field and OPCODE will change, as well as a few other small field changes to block headers. Almost everything else will be internal changes, most of which will be in the EVM layer. Some advanced users may care about changes to some OPCODEs, but those users will be few and far between.”*
```

> *“We will need to add a few new block tags. We currently use `"latest"`, `"earliest"`, and `"pending"`. We may add `"safe"`, `"finalized"`, and probably `"unsafe"` to this list, though we aren't yet sure the what significance it will have outside of forking. The "difficulty" field and OPCODE will change, as well as a few other small field changes to block headers. Almost everything else will be internal changes, most of which will be in the EVM layer. Some advanced users may care about changes to some OPCODEs, but those users will be few and far between.”*

Reviewer 2: ```suggestion
*“We will need to add a few new block tags. We currently use `"latest"`, `"earliest"`, and `"pending"`. We may add `"safe"`, `"finalized"`, and probably `"unsafe"` to this list, though we aren't yet sure of the significance these will have outside of forking. The `difficulty` field and OPCODE will change, as well as a few other small changes to block headers. Almost everything else will be internal changes, most of which will be in the EVM layer. Some advanced users may care about changes to some OPCODEs, but those users will be few and far between.”*
```

Reviewer 2: sorry about this second edit :-)
